### PR TITLE
Credits

### LINKED ISSUE #37: A view to show project contributors
A simple table view that shows all contributors of VMD. A design similar to `CountiesSelectionViewController`.

Linked with https://github.com/CovidTrackerFr/vitemadose/issues/228

### PR DESCRIPTION
**Pull request by NathanFallet**
## Description
I'm adding a view to show contributors, based on the backend contributors endpoint

## Related issue
Fixes #37 

## What I tested
The view itself

## Regression risks
N/A

## Screenshots
_If applicable, add screenshots to help explain the fix. Otherwise please remove this section_
36045A3B-7D95-46A8-A707-97146A77A3D1-9891-000006EB55F5A69D

## Checklist
- [x] I have installed SwiftLint and made sure code formatting is correct
- [x] I have performed a self-review of my own code
- [x] I tested my changes on real device
- [x] My changes generate no new warnings
- [x] I have commented my code in hard-to-understand areas
- [x] Any dependent changes have been merged into **develop**
- [x] I have checked there aren't other open [Pull Requests](https://github.com/CovidTrackerFr/vitemadose-ios/pulls) for the same update/change

### CODE REVIEW

**Comment by pylapp** [comment]
I tested your _credits_ branch, here are some comments for the feature :)

## About the contributors
1. [] For some contributors, we have a pseudonym of the fulll name, some details about contrebutions or not, and an avatar or not. Is it expected?
2. [] How are sorted the contributors? Maybe be should sort them by alphabetic order and by platforms
3. [] For **Paul Jannot**, an error occurred with it hyperlink. The hyperlink _pauljeannot.github.io_ does not work anymore.
4. [] Same thing for **CorentiOS** with its hyperlink _corentinmedina.fr_
5. [] Same thing for **Alexandre GUY** with its hyperlink _www.agraphik.fr_
6. [] Same thing for **Guillaume Rozier** with its hyperlink _guillaumerozier.fr_

## About the UI
1. [] We cannot see easily if we have a clickable cell because an hyperlink is defined, or not.
Maybe we should add in the contributor cell an image button (containing an SF Symbol) which will redirect the user to the hyperlink target on tap

## About a11y
1. [] We may setup a trick for vocalization of "ios" ; maybe convert the word to "iOS" or change the accessible label
2. [] Accessible fonts are not used in this view ; if we use bigger text sizes in device settings, the text size are not changed

## Miscellaneous
1. [] Need to rebase your branch from _develop_ :)

File: ViteMaDose/Views/Credits/Cells/CreditCell.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @IBOutlet private var creditImageView: UIImageView!
+    @IBOutlet private var creditNameLabel: UILabel!
+    @IBOutlet weak var creditRoleLabel: UILabel!
+    @IBOutlet private var creditContainerView: UIView!
+
+    private enum Constant {
+        static let creditNameTextColor: UIColor = .label
+        static let creditImageBackgroundColor: UIColor = .royalBlue
+        static let cellBackgrounColor: UIColor = .tertiarySystemBackground
+        static let labelMainFont: UIFont = .rounded(ofSize: 18, weight: .bold)
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Cells/CreditCell.swift:33]: Should be replaced by:
```swift
        static let labelMainFont: UIFont = UIFont.accessibleCalloutBold
```

File: ViteMaDose/Views/Credits/Cells/CreditCell.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        static let labelSecondFont: UIFont = .rounded(ofSize: 15, weight: .regular)
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Cells/CreditCell.swift:34]: Should be replace by:

```swift
static let labelSecondFont: UIFont = UIFont.accessibleSubheadMedium
```

File: ViteMaDose/Views/Credits/CreditViewController.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    func numberOfSections(in tableView: UITableView) -> Int {
+        viewModel.numberOfSections
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        viewModel.numberOfRows(in: section)
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        // Case of user
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/CreditViewController.swift:79]: Relevant comment? Not sure :)

File: ViteMaDose/Views/Credits/Header/CreditHeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+class CreditHeaderView: UIView {
+    @IBOutlet var titleLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+
+    private enum Constant {
+        static let highlightedTextColor1 = UIColor.royalBlue
+        static let highlightedTextColor2 = UIColor.mandy
+        static let titleFont = UIFont.rounded(ofSize: 24, weight: .bold)
+        static let descriptionFont = UIFont.rounded(ofSize: 18, weight: .regular)
+        static let searchBarFont = UIFont.rounded(ofSize: 16, weight: .medium)
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Header/CreditHeaderView.swift:19]: Unused, should be removed

File: ViteMaDose/Views/Credits/Header/CreditHeaderView.swift [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        static let searchBarViewCornerRadius: CGFloat = 10.0
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Header/CreditHeaderView.swift:20]: Unused, should be removed

File: ViteMaDose/Views/Credits/Header/CreditHeaderView.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import UIKit
+
+class CreditHeaderView: UIView {
+    @IBOutlet var titleLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+
+    private enum Constant {
+        static let highlightedTextColor1 = UIColor.royalBlue
+        static let highlightedTextColor2 = UIColor.mandy
+        static let titleFont = UIFont.rounded(ofSize: 24, weight: .bold)
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Header/CreditHeaderView.swift:17]: Replace by:

```swift
static let titleFont = UIFont.accessibleTitle2Bold
static let descriptionFont = UIFont.accessibleCalloutMedium
```

File: ViteMaDose/Views/Credits/Header/CreditSectionView.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
+//
+//  CreditSectionView.swift
```
Comment (pylapp) [inline ViteMaDose/Views/Credits/Header/CreditSectionView.swift:2]: Are you sure _CreditSectionView_ is used? I don't think so ; these objects should be deleted.

**Review by pylapp — CHANGES_REQUESTED** [review]

**Comment by pylapp** [comment]
@NathanFallet Almost done! The main things to review are about the accessibility (not managed yet in your feature), and also the ordering of the contributors. A bit of dead code too.

**Comment by NathanFallet** [comment]
> For some contributors, we have a pseudonym of the fulll name, some details about contributions or not, and an avatar or not. Is it expected?
Yes. The data is returned by the API at `https://vitemadose.gitlab.io/vitemadose/contributors_all.json`, so we have no control about it in the app, and for some of them not all data is filled.

> For Paul Jannot, an error occurred with it hyperlink. The hyperlink pauljeannot.github.io does not work anymore.
> Same thing for CorentiOS with its hyperlink corentinmedina.fr
> Same thing for Alexandre GUY with its hyperlink www.agraphik.fr
> Same thing for Guillaume Rozier with its hyperlink guillaumerozier.fr
As the data is returned by the API, they need to correct the link where it was fetched from. (I think it was taken from GitHub API, so they need to fix the link in their GitHub profile)

> Need to rebase your branch from develop :)
I started the process, it would take too long to do it, with all the changes since the original branch.

> About KingFisher, register this library in the project (licenses folder) and the app (LicenseInfo.plist)
I let you work on that, you seemed motivated to, and know how to.

**Comment by pylapp** [comment]
@NathanFallet I don't have completed this review and refactoring yet, but it's ongoing.
I hope to complete it this night or tomorrow.

I shared to mates the issue about not working hyperlink and doublons in the contributors JSON file.

Here are the things I will do a soon as possible:
- [ ] Improve VoiceOver vocalization (hint for _ios_, hint for hyperlink icons)
- [ ] Improve UI if too big text size in use
- [ ] Do the rebase from _deveelop_ to keep git history clean
- [ ] Add _KingFisher_ in the project (details in app settings, licence in project)

I keep you in touch !

**Review by pylapp — APPROVED** [review]
Great!

**Pull request merged**